CarRentalapplication/CarRentalApplication.DB/Interfaces/Services/ICarService.cs

```diff
@@ -0,0 +1,14 @@
+﻿using CarRentalApplication.DB.Models;
+using CarRentalApplication.DB.Models.Cars;
+
+namespace CarRentalApplication.DB.Interfaces.Services;
+
+public interface ICarService
+{
+    Task<Car?> GetByIdAsync(int id);
+    Task<IEnumerable<Car>> GetAllAsync();
+    Task<IEnumerable<Car>> GetFilteredAsync(CarFilteringQueryParameters query);
```

Comment:
Let's just have one of these methods only, we don't generally expose much both API's

```suggestion
    Task<IEnumerable<Car>> GetAllAsync(CarFilteringQueryParameters query);
```